Source/Core/Core/ActionReplay.cpp

```diff
@@ -586,10 +586,12 @@ static bool ZeroCode_FillAndSlide(const u32 val_last, const ARAddr& addr, const
   return true;
 }
 
-// Looks like this is new?? - untested
+// kenobi's "memory copy" Z-code. Requires an additional master code
+// on a real AR device. Documented here:
+// https://github.com/dolphin-emu/dolphin/wiki/GameCube-Action-Replay-Code-Types#type-z4-size-3--memory-copy
 static bool ZeroCode_MemoryCopy(const u32 val_last, const ARAddr& addr, const u32 data)
 {
-  const u32 addr_dest = val_last | 0x06000000;
+  const u32 addr_dest = val_last & ~0x06000000;
```

Comment:
This bug must have been due to a misinterpretation of the spec. The 0x06000000 mask needs to be applied by the user, not the implementation (which needs to remove it).

Reply:
Feel free to leave this as a comment above the code snippet (explaining _why_ the mask is here, not _that the code is removing the mask_ which we can see from the code itself).
Right now it is either guesswork or knowing that such a spec exists; but someone with neither cannot understand (or reason about) why this is happening here.

Reply:
Yep, I added a comment above explaining and linking to the spec. A comment here would just be redundant, since it's implied that the bits removed here are not part of the address because they are part of the control bits which activate this command.